yt_dlp/extractor/dplay.py

```diff
@@ -389,3 +389,23 @@ class ScienceChannelIE(DiscoveryPlusIE):
 
     _PRODUCT = 'sci'
     _API_URL = 'us1-prod-direct.sciencechannel.com'
+
+
+class DIYNetworkIE(DiscoveryPlusIE):
+    _VALID_URL = r'https?://(?:www\.|watch\.)?diynetwork\.com/video' + DPlayIE._PATH_REGEX
+    _TESTS = [{
```

Comment:
Add an `only_matching` test for `www.` version. When you do, the test will fail. You'll need to remove that URL pattern from the `ScrippsNetworksIE` to fix it

Reply:
honestly, I don't think it needs it. It doesn't seem to use the www. for anything with media attached to it. Everything that uses that prefix is under "videos" not "video" and still works with the current python scripts. So I'll just take it out and then it should be good right?

Reply:
No. The issue is that there is another extractor catching the `www.` URL. You need to remove it

Reply:
I can add an only_matching test but I can't find a single video that uses `www.diynetwork.com\video\SOMETHING` so I'm not sure how to do that. But I also noticed the scrippsnetwork `valid_url` is looking in the subdirectory `videos` and this is looking in the subdirectory `video`. I thought that since I can't really find an example where the `www` URL would be used in conjunction with `video` and not `videos` it might be best to just take it out of the option for `valid_url` in this script.

ScrippsNetwork.py
```
class ScrippsNetworksIE(InfoExtractor):
    _VALID_URL = r'https?://(?:www\.)?(?P<site>cookingchanneltv|discovery|(?:diy|food)network|hgtv|travelchannel)\.com/videos/[0-9a-z-]+-(?P<id>\d+)'
```

Also yt-dlp can't see the following as a `valid_url` if I take the url pattern out of `ScrippsNetworkIE` but leave the dplay.py script as it is here.

```
__main__.py -F "https://www.diynetwork.com/videos/barnwood-chalkboard-key-holder-0265789" --verbose
[debug] Command-line config: ['-F', 'https://www.diynetwork.com/videos/barnwood-chalkboard-key-holder-0265789', '--verbose']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.08.10 (source)
[debug] Plugin Extractors: ['SamplePlugin']
[debug] Git HEAD: 8c59e774b
[debug] Python version 3.9.4 (CPython 64bit) - Windows-10-10.0.18362-SP0
[debug] exe versions: ffmpeg 4.4-full_build-www.gyan.dev, ffprobe 4.4-full_build-www.gyan.dev
[debug] Optional libraries: mutagen, pycryptodome, sqlite, websockets
[debug] Proxy map: {}
[debug] [generic] Extracting URL: https://www.diynetwork.com/videos/barnwood-chalkboard-key-holder-0265789
[generic] barnwood-chalkboard-key-holder-0265789: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] barnwood-chalkboard-key-holder-0265789: Downloading webpage
[generic] barnwood-chalkboard-key-holder-0265789: Extracting information
ERROR: [generic] Unsupported URL: https://www.diynetwork.com/videos/barnwood-chalkboard-key-holder-0265789
Traceback (most recent call last):
  File "D:\yt-dlp\yt_dlp\extractor\common.py", line 584, in extract
    ie_result = self._real_extract(url)
  File "D:\yt-dlp\yt_dlp\extractor\generic.py", line 3713, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.diynetwork.com/videos/barnwood-chalkboard-key-holder-0265789
Traceback (most recent call last):
  File "D:\yt-dlp\yt_dlp\extractor\common.py", line 584, in extract
    ie_result = self._real_extract(url)
  File "D:\yt-dlp\yt_dlp\extractor\generic.py", line 3713, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.diynetwork.com/videos/barnwood-chalkboard-key-holder-0265789

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\yt-dlp\yt_dlp\YoutubeDL.py", line 1214, in wrapper
    return func(self, *args, **kwargs)
  File "D:\yt-dlp\yt_dlp\YoutubeDL.py", line 1239, in __extract_info
    ie_result = ie.extract(url)
  File "D:\yt-dlp\yt_dlp\extractor\common.py", line 600, in extract
    raise ExtractorError(
yt_dlp.utils.ExtractorError: [generic] Unsupported URL: https://www.diynetwork.com/videos/barnwood-chalkboard-key-holder-0265789
```

Reply:
It is my mistake. I was thinking `video` and `videos` URLs to be the same. But since their extraction process seems to be totally different, the current code makes sense